x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/QueryStringFunction.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function.fulltext;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.expression.Foldables;
+import org.elasticsearch.xpack.esql.core.querydsl.query.Query;
+import org.elasticsearch.xpack.esql.core.querydsl.query.QueryStringQuery;
+import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.expression.function.Example;
+import org.elasticsearch.xpack.esql.expression.function.FunctionInfo;
+import org.elasticsearch.xpack.esql.expression.function.Param;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Full text function that performs a {@link QueryStringQuery} .
+ */
+public class QueryStringFunction extends FullTextFunction {
```

Comment:
Usually an ES|QL function has its own `AbstractFunctionTestCase`, an important usage is to generate documents(some of them are used by Kibana) automatically. This function doesn't have its compute engine implementation, I'm not quite sure how to make an `AbstractFunctionTestCase` for it yet, need to figure out how to generate the docs automatically, or manually create them.

Reply:
`QueryStringFunction` takes only string literal, however if a numeric type is provided, it passes `Analyzer`, if a field is provided as input, it fails at ExpressionTranslator, these could be candidates for `VerifierTests`.

Override `resolveType()` is the usual way to validate input data types and foldable for functions during `Analyzer` phase, perhaps refer to an ES|QL string function as an example.

Reply:
Good point on  `AbstractFunctionTestCase`, but we don't want generate documentation yet since the function is only available in snapshots. Also I might be looking at the wrong place, but we have another function `Rate` that is only available in snapshots and it also seems to be missing its own `AbstractFunctionTestCase`.
So I wonder if we can defer the need for a `AbstractFunctionTestCase` for now, since it seems it requires a bit of rework so that `AbstractFunctionTestCase` can also work with functions that do not have a compute engine implementation.

Reply:
> QueryStringFunction takes only string literal, however if a numeric type is provided, it passes Analyzer, if a field is provided as input, it fails at ExpressionTranslator, these could be candidates for VerifierTests.
> 
> Override resolveType() is the usual way to validate input data types and foldable for functions during Analyzer phase, perhaps refer to an ES|QL string function as an example.

I've tried to address this in 7b2cbaa151cd60b6ba71a8aeb541463ef59ad344

Reply:
++ Ioana on documentation and `AbstractFunctionTestCase` - we can probably defer this until we plan to move it out of snapshot builds

Reply:
I added a ui label, so that Kibana is aware of this. As far as I know `AbstractFunctionTestCase` generates docs for Kibana, we can figure out the best option to move forward.

Reply:
> ++ Ioana on documentation and `AbstractFunctionTestCase` - we can probably defer this until we plan to move it out of snapshot builds

Can an issue be created to keep track of `AbstractFunctionTestCase` if it won't be included in this PR? It is important when this function is taken out of snapshot, we may not want to loose track of it. Thank you!

Reply:
I've actually given it a go [here](https://github.com/elastic/elasticsearch/pull/112590/files#diff-bd42b7827384b735ff62baeb045ff9e37edcec201b5ee72f026e1cdb8c6496e8) in 4b97e6da93f1ad9d1e763b398ecb43befec45b30 - it performs type checking and folding.